warp/controller/warpMenuConfigReconciler.go

```diff
@@ -0,0 +1,143 @@
+package controller
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/cloudogu/warp-assets/config"
+	"github.com/cloudogu/warp-assets/controller/types"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	types2 "k8s.io/apimachinery/pkg/types"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/event"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+)
+
+const (
+	globalConfigMapName              = "global-config"
+	warpMenuUpdateEventReason        = "WarpMenu"
+	errorOnWarpMenuUpdateEventReason = "ErrUpdateWarpMenu"
+)
+
+type WarpMenuConfigReconciler struct {
+	client              k8sClient
+	globalConfigRepo    GlobalConfigRepository
+	doguVersionRegistry DoguVersionRegistry
+	localDoguRepo       LocalDoguRepo
+	eventRecorder       eventRecorder
+	warpMenuPath        string
+	deploymentName      string
+}
+
+func NewWarpMenuReconciler(client k8sClient, globalConfigRepo GlobalConfigRepository, doguVersionRegistry DoguVersionRegistry, localDoguRepo LocalDoguRepo, eventRecoder eventRecorder, warpMenuPath string, deploymentName string) *WarpMenuConfigReconciler {
+	return &WarpMenuConfigReconciler{
+		client:              client,
+		globalConfigRepo:    globalConfigRepo,
+		doguVersionRegistry: doguVersionRegistry,
+		localDoguRepo:       localDoguRepo,
+		eventRecorder:       eventRecoder,
+		warpMenuPath:        warpMenuPath,
+		deploymentName:      deploymentName,
+	}
+}
+
+func (r *WarpMenuConfigReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	logger := log.FromContext(ctx)
+	logger.Info("WarpMenuConfigReconciler reconcile()")
+
+	deployment := &appsv1.Deployment{}
+	err := r.client.Get(ctx, types2.NamespacedName{Name: r.deploymentName, Namespace: req.Namespace}, deployment)
+	if err != nil {
+		return ctrl.Result{}, fmt.Errorf("warp update: failed to get deployment [%s]: %w", "k8s-ces-assets-nginx", err)
+	}
+
+	warpMenuConfiguration, err := config.ReadConfiguration(ctx, r.client, req.Namespace)
+	if err != nil {
+		r.eventRecorder.Eventf(deployment, corev1.EventTypeWarning, errorOnWarpMenuUpdateEventReason, "Reading warp menu config failed: %w", err)
```

Comment:
Is it even a good idea to write events to the deployment instead of just simply logging? I mean usually we use events to write events to a custom resource we watch.